src/lib.rs

```diff
@@ -25,11 +25,101 @@
 //!   with a `Mutex`. Ultimately the goal is to include stacks, queues, deques,
 //!   bags, sets and maps. These subcrates are reexported in the `sync` module.
 
-//#![deny(missing_docs)]
+#![warn(missing_docs)]
 
-pub extern crate crossbeam_epoch as epoch;
-pub extern crate crossbeam_utils as utils;
+extern crate crossbeam_channel;
+extern crate crossbeam_deque;
+extern crate crossbeam_epoch;
+extern crate crossbeam_utils;
 
-pub mod sync;
-pub use utils::cache_padded::CachePadded;
-pub use utils::scoped::{scope, Scope, ScopedJoinHandle, ScopedThreadBuilder};
+mod arc_cell;
+mod ms_queue;
+mod seg_queue;
+mod treiber_stack;
+
+
+/// Additional utilities for atomics.
+pub mod atomic {
+    pub use crossbeam_utils::AtomicConsume;
+    pub use arc_cell::ArcCell;
+    // This module will grow in the future.
+}
+
+/// Scoped thread.
+///
+/// See [the `crossbeam-utils` crate](https://github.com/crossbeam-rs/crossbeam-utils) for more
+/// information.
+pub mod thread {
+    pub use crossbeam_utils::thread::*;
+}
+
+// Export `scope` and `CachePadded` in the crate root in order not to break established patterns.
+pub use thread::scope;
+pub use crossbeam_utils::CachePadded;
```

Comment:
`crossbeam::CachePadded` is not that common (in contrast to `crossbeam::scope`, which everybody knows about) so in this case I'd prefer to simply leave `CachePadded` inside `utils` module only. What do you think?

Reply:
> Since `crossbeam-utils` is our "catch-all" crate for random things that don't deserve their own crates (or at least not yet), why not scatter its contents around more logically?

I agree with your statement above, so I'm hesitant to introduce `utils` submodule in `crossbeam`.  Maybe someday we will introduce a submodule to which `CachePadded` logically belongs, but for now, I cannot imagine what it will be...  That's why I'd like to put `CachePadded` at the root for the moment.

I think crate organization is an important matter, so even though this PR is approved, I'll wait for us to agree on this matter.

Reply:
Another contender for `utils` I've often wanted is [`Backoff`](https://github.com/crossbeam-rs/crossbeam-channel/blob/ce0b2db0ef5a3609cb6e628e3aff649e8eaf4860/src/internal/utils.rs#L13) struct (also really difficult to put into a more specific category). It automatically yields the processor/thread in CAS loop.